sdk/storage/storage-blob/review/generated/src/models/index.d.ts

```diff
@@ -0,0 +1,6713 @@
+     */
+    signedService: string;
+    /**
+     * The service version that created the key
+     */
+    signedVersion: string;
+    /**
+     * The key as a base64 string
+     */
+    value: string;
+}
+/**
+ * An interface representing StorageError.
+ */
+export interface StorageError {
+    message?: string;
+}
+/**
+ * The service error response object.
+ */
+export interface DataLakeStorageErrorError {
+    /**
+     * The service error code.
+     */
+    code?: string;
+    /**
+     * The service error message.
+     */
+    message?: string;
+}
+/**
+ * An interface representing DataLakeStorageError.
+ */
+export interface DataLakeStorageError {
+    /**
+     * The service error response object.
+     */
+    error?: DataLakeStorageErrorError;
+}
+/**
+ * An Access policy
+ */
+export interface AccessPolicy {
+    /**
+     * the date-time the policy is active
+     * **NOTE: This entity will be treated as a string instead of a Date because the API can
+     * potentially deal with a higher precision value than what is supported by JavaScript.**
+     */
+    start: string;
+    /**
+     * the date-time the policy expires
+     * **NOTE: This entity will be treated as a string instead of a Date because the API can
+     * potentially deal with a higher precision value than what is supported by JavaScript.**
+     */
+    expiry: string;
+    /**
+     * the permissions for the acl policy
+     */
+    permission: string;
+}
+/**
+ * Properties of a blob
+ */
+export interface BlobProperties {
+    creationTime?: Date;
+    lastModified: Date;
+    etag: string;
+    /**
+     * Size in bytes
+     */
+    contentLength?: number;
+    contentType?: string;
+    contentEncoding?: string;
+    contentLanguage?: string;
+    contentMD5?: Uint8Array;
+    contentDisposition?: string;
+    cacheControl?: string;
+    blobSequenceNumber?: number;
+    /**
+     * Possible values include: 'BlockBlob', 'PageBlob', 'AppendBlob'
+     */
+    blobType?: BlobType;
+    /**
+     * Possible values include: 'locked', 'unlocked'
+     */
+    leaseStatus?: LeaseStatusType;
+    /**
+     * Possible values include: 'available', 'leased', 'expired', 'breaking', 'broken'
+     */
+    leaseState?: LeaseStateType;
+    /**
+     * Possible values include: 'infinite', 'fixed'
+     */
+    leaseDuration?: LeaseDurationType;
+    copyId?: string;
+    /**
+     * Possible values include: 'pending', 'success', 'aborted', 'failed'
+     */
+    copyStatus?: CopyStatusType;
+    copySource?: string;
+    copyProgress?: string;
+    copyCompletionTime?: Date;
+    copyStatusDescription?: string;
+    serverEncrypted?: boolean;
+    incrementalCopy?: boolean;
+    destinationSnapshot?: string;
+    deletedTime?: Date;
+    remainingRetentionDays?: number;
+    /**
+     * Possible values include: 'P4', 'P6', 'P10', 'P15', 'P20', 'P30', 'P40', 'P50', 'P60', 'P70',
+     * 'P80', 'Hot', 'Cool', 'Archive'
+     */
+    accessTier?: AccessTier;
```

Comment:
There a lot of enums here. What happens if the service returns a value that is not in the valid list? This is very possible in the future.

Reply:
I suppose we will re-generated this code on the updated swagger api definitions corresponding to the new service version.

Reply:
`AccessTier` is a union type of string literals
```ts
export type AccessTier = 'P4' | 'P6' | 'P10' | 'P15' | 'P20' | 'P30' | 'P40' | 'P50' | 'P60' | 'P70' | 'P80' | 'Hot' | 'Cool' | 'Archive';
```

Reply:
We will re-generate and update swagger for every new storage API version and new tiers.

In the runtime, if mentioned senario happens, there should not have errors (if protocol layer deserializer doesn't check enum values). In JS there is no enum. The property is actually a string.